v1.1/parallel-statement-execution.md

```diff
@@ -0,0 +1,128 @@
+
+### Error message mismatch
+
+In case of parallel statement execution, it is possible to see a mismatch in the error message being displayed and the statement being executed. 
+
+Because the `RETURNING NOTHING` clause is used with parallel statements, CockroachDB cannot return the error for the parallel statement. The error message for the parallel statement is then reported on the next barrier statement. Thus you might see a mismatch with the error message to the statement being executed. 
+
+On encountering the error, CockroachDB will stop the execution of all parallel statements and [abort and retry the transaction](transactions.html#error-handling). So even an error message mismatch occurs, the result will be the same as aborting the transaction.
+
+Revisiting our sample scenario, suppose an error occurs while executing the second UPDATE statement. In sequential execution, the transaction will be aborted and the error message will be sent immediately. But in case of parallel execution, the transaction will be aborted and error message will be sent after COMMIT. The following diagram illustrates the concept of error message mismatch:
+
+<img src="{{ 'images/Parallel_Statement_Execution_Error_Mismatch.png' | relative_url }}" alt="CockroachDB Parallel Statement Execution Error Mismatch" style="border:1px solid #eee;max-width:100%" />
+
+### `RETURNING NOTHING` clause appended to dependent statements
+
+If two consecutive statements are not independent, and yet a `RETURNING NOTHING` clause is added to the statements, CockroachDB will detect the dependence and serialize the execution. You can thus use the `RETURNING NOTHING` clause with SQL statements without worrying about their dependence.
+
+Revising our sample scenario, suppose we want to create a new user on the social networking app. We need to create entries for the last name of the user, their favorite movie, and favorite song. We need to insert entries into three tables: username, movies, and songs. The transaction would be as follows:
+
+~~~ sql
+> BEGIN;
+> id = 2;
+> INSERT INTO username VALUES last_name = 'Pavlo' WHERE id = 2 RETURNING NOTHING;
+> INSERT INTO movies VALUES favorite_movie = 'Godfather' WHERE id = 2 RETURNING NOTHING;
+> INSERT INTO songs VALUES favorite_song = 'Remember' WHERE id = 2 RETURNING NOTHING;
+> COMMIT;
+~~~
+
+In this case, the second and third INSERT statements are dependent on the first INSERT statement. Thus even though we appended the RETURNING NOTHING clause to the first statement, CockroachDB will execute the first statement sequentially. After the first statement is executed, the second and third statements will be executed in parallel. The following conceptual diagram shows how the transaction will be executed in sequential and parallel modes:
```

Comment:
> are dependent on the first INSERT statement

Add to this "because the `movies` and `songs` tables both have a foreign key constraint on the `username` table".